libvuln/updatesets.go

```diff
@@ -0,0 +1,80 @@
+package libvuln
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/quay/claircore/alpine"
+	"github.com/quay/claircore/aws"
+	"github.com/quay/claircore/debian"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/oracle"
+	"github.com/quay/claircore/photon"
+	"github.com/quay/claircore/suse"
+	"github.com/quay/claircore/ubuntu"
+	"github.com/rs/zerolog"
+)
+
+var defaultSets = map[string]func() (driver.UpdaterSet, error){
+	"alpine": alpine.UpdaterSet,
+	"aws":    aws.UpdaterSet,
+	"debian": debian.UpdaterSet,
+	"oracle": oracle.UpdaterSet,
+	"photon": photon.UpdaterSet,
+	"suse":   suse.UpdaterSet,
+	"ubuntu": ubuntu.UpdaterSet,
+}
+
+// UpdaterSets returns all UpdaterSets currently
+// supported by libvuln
+func updaterSets(ctx context.Context, sets *[]string) (driver.UpdaterSet, error) {
+	log := zerolog.Ctx(ctx).With().
+		Str("component", "libvuln/updaterSets").
+		Logger()
+
+	us := driver.NewUpdaterSet()
```

Comment:
This can be restructured as `if sets == nil { sets = defaults }` and then a single loop merging the sets.

Reply:
defaultSets is a map of constructors, errors still need to be checked for each.